examples/Makefile

```diff
@@ -13,8 +13,12 @@ release:
 	$(CP) -r ./kafka-mirror-maker $(RELEASE_PATH)/
 	$(CP) -r ./kafka-mirror-maker-2 $(RELEASE_PATH)/
 	$(CP) -r ./kafka-bridge $(RELEASE_PATH)/
+	$(CP) -r ./connector $(RELEASE_PATH)/
+	$(CP) -r ./cruise-control $(RELEASE_PATH)/
+	$(CP) -r ./security $(RELEASE_PATH)/
 	$(CP) -r ./user $(RELEASE_PATH)/
 	$(CP) -r ./topic $(RELEASE_PATH)/
 	$(CP) -r ./metrics $(RELEASE_PATH)/
+	$(CP) -r ./README.md $(RELEASE_PATH)/
```

Comment:
Can't we write it to just copy everything (or blacklist what we don't want), rather than having to whitelist everything we do want?

Reply:
I'm in agreement with Tom here. And if you want a finer audit of what exactly that means would be going into the distribution archive then as part of the  build we could ls the path and print it to a file that would act as a manifest so come release time we know with relative certainty what has creeped in or left

Reply:
There is no exclude in copy command AFAIK and we need to exclude at least the Makefile file and I really prefer to keep things readable rather than build sophisticated complex constructs.

Reply:
I believe you can blacklist with the `find` command and then pipe the filtered list to `cp` - but you are right this may be needlessly complicated. Maybe a thought for future releases

Reply:
`$(CP) -r $$(ls | $(GREP) -Ev '^Makefile$$') $(RELEASE_PATH)/`

Reply:
I'm sure that works. But I would still prefer to have commands which I actually understand and know what they do when I run into problems.

Reply:
Because this is in a Makefile `$(...)` is interpreted as replacement by `make` parameters (e.g. `$(CP)`, as you already know). You have to use `$$` to generate a literal `$`, so the `$$(...)` construct just means that `$(...)` gets passed to the shell to execute, and that's just a subshell (as you already know). Similarly what would be `$` in the regex to match the end of input is quoted to `$$`.

Reply:
Happy?